File: Leetcode/2221 - Find Triangular Sum of an Array/solution.py
# Naive Solution
def triangularSum(nums: List[int]) -> int:
    while len(nums) > 1:
        temp = []
        for i in range(1, len(nums)):
            temp.append((nums[i] + nums[i - 1]) % 10)
        nums = temp
    return nums[0]

# Better Solution
def triangularSum(self, nums: List[int]) -> int:
    n = len(nums)
    while n > 1:
        for i in range(1, len(nums)):
            nums[i - 1] = (nums[i - 1] + nums[i]) % 10
        n -= 1
    return nums[0]

File: Leetcode/2221 - Find Triangular Sum of an Array/solution.cpp
// Naive Solution
int triangularSum(vector<int>& nums) {
    while (nums.size() > 1) {
        vector<int> temp;
        for (int i = 1; i < nums.size(); i++) {
            temp.push_back((nums[i] + nums[i-1]) % 10);
        }
        nums = temp;
    }
    return nums[0];
}

// Better Solution
int triangularSum(vector<int>& nums) {
    int n = nums.size();
    while (n > 1) {
        for (int i = 1; i < nums.size(); i++) {
            nums[i-1] = (nums[i-1] + nums[i]) % 10; 
        }
        n--;
    }
    return nums[0];
}

// Best Solution
// Complex to code due to Signed Integer Overflow issue for the given constraints
// Explained in solutions